src/msg/async/rdma/RDMAConnectedSocketImpl.cc

```diff
@@ -233,18 +239,19 @@ void RDMAConnectedSocketImpl::handle_connection() {
         ldout(cct, 10) << __func__ << " server is already active." << dendl;
         return ;
       }
+      r = activate();
+      assert(!r);
       r = infiniband->send_msg(cct, tcp_fd, my_msg);
       if (r < 0) {
         ldout(cct, 1) << __func__ << " server ack failed." << dendl;
         dispatcher->perf_logger->inc(l_msgr_rdma_handshake_errors);
         fault();
         return ;
       }
-      r = activate();
-      assert(!r);
     } else { // ack from client
       connected = 1;
-      cleanup();
+      ldout(cct, 10) << __func__ << " handshake of rdma is done. server connected: " << connected << dendl;
+      //cleanup();
```

Comment:
1. client send "FIN M" msg to server to disconnect 
2. after server recieve "FIN M" msg, it enters CLOSE_WAIT state and send "ACK M+1" msg to client
3. client comes in FIN_WAIT2 stat after it recieve "ACK M+1" msg
4. server send "FIN N" msg to client when it has processed all its task, and make itself in LAST_ACK state
5. client accesses TIME_WAIT state when it recieves "FIN N", and send "ACK N+1" msg to server.
6. server accesses CLOSED state with "ACK N+1" msg recieved.

According to the above process， CLOSE_WAIT will only appears on the server when "ACK M+1" msg has not been sent by server.
CLOSE_WAIT will cause server socket to be heavily occupied.